openHarmony/openHarmony_file.js

```diff
@@ -324,16 +324,26 @@ $.oFolder.prototype.create = function(){
  * @param   {bool}     [overwrite]           Whether to overwrite the target.
  */
 $.oFolder.prototype.copy = function( folderPath, copyName, overwrite ){
+    // TODO: right now it wont overwrite any files. It copies the missing folders and files and leave the existing ones untouched
+    // TODO: it should propagate errors from the recursive copy and throw them before ending?
     if (typeof overwrite === 'undefined') var overwrite = false;
     if (typeof copyName === 'undefined') var copyName = this.name;
-    if (typeof folderPath === 'undefined') var folderPath = this.folder.path;
-
+    if (!(folderPath instanceof this.$.oFolder)) folderPath = new $.oFolder(folderPath);
     if (this.name == copyName && folderPath == this.folder.path) copyName += "_copy";
-
     var copyPath = folderPath+copyName;
 
-    // TODO: deep recursive copy file by file of the contents
-
+    var nextFolder = new $.oFolder(folderPath.path+"/"+this.name);
```

Comment:
here, can we change `this.name` to `copyName`, in case our user wants to rename its copy?